.github/workflows/publish.yml

```diff
@@ -0,0 +1,60 @@
+name: Update Version, Test, Publish to PyPI, and Upload Release Artifacts
+
+on:
+  release:
+    types: [created, edited]
+
+jobs:
+  build-test-publish-upload:
+    runs-on: ubuntu-latest
+    environment: release
+    permissions:
+      # IMPORTANT: this permission is mandatory for trusted publishing
+      id-token: write
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install hatch
+    - name: Set release version
+      run: |
+        # Get the tag from the GitHub release
+        TAG=${GITHUB_REF#refs/tags/}
+        # Remove 'v' prefix if present
+        VERSION=${TAG#v}
+        hatch version $VERSION
+    - name: Build package
+      run: hatch build
+    - name: Run tests
+      run: hatch run test:pytest
+    - name: Publish package to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+    - name: Verify PyPI Release
+      run: |
+        # Verify PyPI release
+        PACKAGE_NAME="mesa_frames"
+        CURRENT_VERSION=$(hatch version)
+        pip install $PACKAGE_NAME==$CURRENT_VERSION
+        python -c "import mesa_frames; print(mesa_frames.__version__)"
+    - name: Update to Next Version
```

Comment:
@EwoutH we should do this in core Mesa as well.

Reply:
This is quite slick indeed.

Maybe @adamamer20 can take a look at our CI process and see if any of those can be improved based on his experience.

Reply:
In the end it would be nice if the workflows were similar between Mesa and Mesa-frames, so this workflow and the [one of Mesa](https://github.com/projectmesa/mesa/blob/main/.github/workflows/release.yml) get aligned with each other.

Reply:
> This is quite slick indeed.
> 
> Maybe @adamamer20 can take a look at our CI process and see if any of those can be improved based on his experience.

Sure, I'd be happy to open some PRs to match mesa-frames's CI